create.ps1

```diff
@@ -1,258 +1,449 @@
-        $CorrelationField = $actionContext.CorrelationConfiguration.accountField
-        $CorrelationValue = $actionContext.CorrelationConfiguration.PersonFieldValue
+    #region Verify correlation configuration and properties
+    $actionMessage = "verifying correlation configuration and properties"
 
-        if ($Null -eq $CorrelationField -or $Null -eq $CorrelationValue) {
+    if ($actionContext.CorrelationConfiguration.Enabled -eq $true) {
+        if ([string]::IsNullOrEmpty($correlationField)) {
             throw "Correlation is enabled but not configured correctly."
         }
-
-        #  Write logic here that checks if the account can be correlated in the target system
-        $SplatParams = @{
-            Uri    = "$($actionContext.Configuration.BaseUrl)/Users"
-            Method = "Get"
-            Body   = @{
-                filter = "$($CorrelationField) eq '$($CorrelationValue)'"
-            }
+        
+        if ([string]::IsNullOrEmpty($correlationValue)) {
+            throw "The correlation value for [$correlationField] is empty. This is likely a mapping issue."
         }
-        $CorrelatedAccount = Invoke-RestMethod @LisaRequest @SplatParams
-
-        if ($CorrelatedAccount.count -gt 1) {
-            throw "Multiple accounts found with filter: $($SplatParams.Body.filter)"
+    }
+    else {
+        Write-Warning "Correlation is disabled."
+    }
+    #endregion Verify correlation configuration and properties
+    
+    #region Create access token
+    $actionMessage = "creating access token"
+    
+    $createAccessTokenBody = @{
+        grant_type    = "client_credentials"
+        client_id     = $actionContext.Configuration.EntraIDAppId
+        client_secret = $actionContext.Configuration.EntraIDAppSecret
+        scope         = $actionContext.Configuration.KPNMWPScope
+    }
+    
+    $createAccessTokenSplatParams = @{
+        Uri             = "https://login.microsoftonline.com/$($actionContext.Configuration.EntraIDTenantID)/oauth2/v2.0/token/"
+        Headers         = $headers
+        Method          = "POST"
+        ContentType     = "application/x-www-form-urlencoded"
+        UseBasicParsing = $true
+        Body            = $createAccessTokenBody
+        Verbose         = $false
+        ErrorAction     = "Stop"
+    }
+    
+    $createAccessTokenResonse = Invoke-RestMethod @createAccessTokenSplatParams
+    
+    Write-Verbose "Created access token. Result: $($createAccessTokenResonse | ConvertTo-Json)"
+    #endregion Create access token
+    
+    #region Create headers
+    $actionMessage = "creating headers"
+    
+    $headers = @{
+        "Authorization"   = "Bearer $($createAccessTokenResonse.access_token)"
+        "Accept"          = "application/json"
+        "Content-Type"    = "application/json;charset=utf-8"
+        "Mwp-Api-Version" = "1.0"
+    }
+    
+    Write-Verbose "Created headers. Result: $($headers | ConvertTo-Json)."
```

Comment:
Ook voor de headers kunnen we het token mogelijk uit de headers laten voordat we dit naar de logging schrijven.

Reply:
Eens, aangepast om resultaat zonder authorization te loggen. Dit is toch cruciale info en loggen we niet bij de acties zelf, aangezien we daar de headers niet meenemen.